piv/key.go

```diff
@@ -667,24 +671,82 @@ func (yk *YubiKey) PrivateKey(slot Slot, public crypto.PublicKey, auth KeyAuth)
 	}
 }
 
-type keyECDSA struct {
+// ECDSAPrivateKey is used to access a ECDSA signing key.
```

Comment:
nit: include user instructions to get a *ECDSAPrivateKey. Anyone reading the docs will probably appreciate that :)

```
// ECDSAPrivateKey is a crypto.PrivateKey implementation used to directly access ECDSA
// operations. Keys returned by PrivateKey() can be type casted to use these methods.
//
//     priv, err := yk.PrivateKey(slot, pub, auth)
//     if err != nil {
//         // ...
//     }
//     ecdsaPriv, ok := priv.(piv. ECDSAPrivateKey)
```

Reply:
I wrote proper examples for SharedKey, that should also demonstrate this, and the documentation for the type referring there.